app/models/employer_events/employer_importer.rb

```diff
@@ -4,31 +4,111 @@ class EmployerImporter
-        fein: fein,
-        dba: dba,
-        name: company_name
+        hbx_id: @org.id,
+        fein: @org.fein,
+        dba: @org.dba,
+        name: @org.name
       }
     end
 
+    def manage_employer_demographics(employer)
+      if employer.employer_contacts.blank? || @event_name == "contact_changed"
+        add_contacts(@org.contacts, employer)
+      end
+
+      if employer.employer_office_locations.blank? ||  @event_name == "address_changed"
+        add_office_locations(@org.office_locations,employer)
+      end
+    end
+
+    def add_contacts(incoming_contacts, employer)
+      employer.employer_contacts = incoming_contacts.map do |incoming_contact| 
+        contact_attributes =  {
+          name_prefix: incoming_contact.name_prefix,
+          first_name: incoming_contact.first_name,
+          middle_name: incoming_contact.middle_name,
+          last_name: incoming_contact.last_name,
+          name_suffix: incoming_contact.name_suffix,
+          job_title: incoming_contact.job_title,
+          department: incoming_contact.department
+        }
+        contact_attributes.delete_if{ |k,v| v.blank?}
+        new_contact = EmployerContact.new(contact_attributes)
+        add_contacts_phones(incoming_contact.phones, new_contact)  
+        add_contacts_addresses(incoming_contact.addresses, new_contact) 
+        add_contacts_emails(incoming_contact.emails, new_contact) 
+        new_contact
+      end
+      employer.save!
+    end
+
+    def add_contacts_phones(incoming_phones, new_contact)
+      new_contact.phones = incoming_phones.map do |incoming_phone|
+        new_phone(incoming_phone)
+      end
+    end
+
+    def add_contacts_emails(incoming_emails, new_contact)
+      new_contact.emails = incoming_emails.map do |incoming_email|
+        new_email(incoming_email)
+      end
+    end
+    
+    def add_contacts_addresses(incoming_addresses, new_contact)
+      new_contact.addresses = incoming_addresses.map  do |incoming_address|
+        new_address(incoming_address)
+      end
+    end
+    
+    def add_office_locations(incoming_office_locations)
+      employer.employer_office_locations = incoming_office_locations.map do |incoming_office_location|   
+          new_location = EmployerOfficeLocation.new(name:incoming_office_location.name) 
+          new_location.phone = new_phone(incoming_office_location.phone)
+          new_location.address = new_address(incoming_office_location.address)
+          new_location
+        end
+      employer.save!
+    end
+
+    def new_email(incoming_email)
+      Email.new(
+        type: incoming_email.type,
+        email_address: incoming_email.email_address
+      )
+    end
+    
+    def new_phone(incoming_phone)
+      Phone.new(
+        phone_number:  incoming_phone.full_phone_number.last(7),
+        full_phone_number: incoming_phone.full_phone_number,
+        phone_type: incoming_phone.type,
+        primary: incoming_phone.is_preferred
+      )
+    end
```

Comment:
Careful here - last 4 of the phone number aren't always nil - sometimes there is an extension.

Also, glue doesn't usually even populate full phone number (!) it usually sticks that value in the "phone_number" field.  Makes perfect sense, right?